Bottom: 503e916a35ea4d509b0fe2100ee590082cf66afa
Top:    56f6a76c01356a2082a366a24b272f1433ed8b61
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-24 03:15:57 +0000

Hack around Circuit CAM / EDG brokeness reading drill files







---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 54c7f4a..dfa7f00 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -52,10 +52,16 @@ static void gerber_fill_polygon (hidGC gc, int n_coords, int *x, int *y);
 #define gerberYOffset(pcb, y) ((long) (-(y)))
 
 /* These are for drills */
-#define gerberDrX(pcb, x) ((long) ((x)/10))
-#define gerberDrY(pcb, y) ((long) (((pcb)->MaxHeight - (y)))/10)
-#define gerberDrXOffset(pcb, x) ((long) ((x)/10))
-#define gerberDrYOffset(pcb, y) ((long) (-(y))/10)
+#ifdef EDG_ARE_BROKEN
+#  define EDG_MUL 100
+#else
+#  define EDG_MUL 1
+#endif
+
+#define gerberDrX(pcb, x) ((long) (EDG_MUL * (x)/10))
+#define gerberDrY(pcb, y) ((long) (EDG_MUL * ((pcb)->MaxHeight - (y))/10))
+#define gerberDrXOffset(pcb, x) ((long) (EDG_MUL * (x) / 10))
+#define gerberDrYOffset(pcb, y) ((long) (EDG_MUL * -(y) / 10))
 
 /*----------------------------------------------------------------------------*/
 /* Private data structures                                                    */
